pkl-tools/pkl-tools.gradle.kts

```diff
@@ -65,6 +66,63 @@ val javadocJar by
 
 tasks.shadowJar { archiveBaseName.set("pkl-tools-all") }
 
+private fun Exec.configureTestStartFatJar(launcher: Provider<JavaLauncher>) {
+  dependsOn(tasks.shadowJar)
+  group = "verification"
+
+  // placeholder output to satisfy up-to-date check
+  val outputFile = layout.buildDirectory.file("testStartFatJar/${name}.txt")
+  outputs.file(outputFile)
+
+  inputs.files(tasks.shadowJar)
+  executable = launcher.get().executablePath.asFile.absolutePath
+
+  argumentProviders.add(
+    CommandLineArgumentProvider {
+      buildList {
+        add("-cp")
+        add(tasks.shadowJar.get().outputs.files.singleFile.absolutePath)
+        add("org.pkl.cli.Main")
+        add("eval")
+        add("-x")
+        add("1 + 1")
+        add("pkl:base")
+      }
+    }
+  )
+
+  doLast {
+    outputFile.get().asFile.let { file ->
+      file.ensureParentDirsCreated()
```

Comment:
Not sure what this is, but it is internal and I wouldn't use it. It's coming from the Kotlin gradle plugin and can be changed with every release.

Reply:
Good catch!